### PR TITLE
fix(doc): update EnvResourceDetector specification link

### PR DESCRIPTION
**Pull request by ymgyt**
update `EnvResourceDetector`  not found link https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#telemetry-sdk 

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by hdost — APPROVED** [review]
Thank you for the find 👍 small suggestion, not mandatory.

File: opentelemetry-sdk/src/resource/env.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@ const OTEL_SERVICE_NAME: &str = "OTEL_SERVICE_NAME";
 /// general SDK configuration from environment.
 ///
 /// See
-/// [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#telemetry-sdk)
+/// [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md)
```
[the remark this answers is not in the document]
Comment (ymgyt) [reply opentelemetry-sdk/src/resource/env.rs]: thanks for suggestion, I have added the anchor.

**Review by shaun-cox — APPROVED** [review]
Thank you!

**Review by cijothomas — APPROVED** [review]
Thanks!
We could run some tool to detect any broken links and do a cleanup to ensure all other links are valid!

**Pull request merged**